### ISSUE TITLE
[HOLD for payment 2024-04-15] [$500]  BA - On the phone number page, user can not enter data as the placeholder shows

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#37723]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** v1.4.47-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** applausetester+vd_web030424@applause.expensifail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team

## Action Performed:
Pre-requisite: user must be logged in and have created a workspace.

1. Go to Workspace > Bank account.
2. Initiate the add BA flow with the testing credentials.
3. On Company information section, reach the Phone number page.
4. Verify the placeholder is displayed as "(xxx)xxx-xxxx".
5. Enter a random US number in that same format, like "(440)458-9784".
6. Click on "Next".
7. Verify an error appears.

## Expected Result:
The user should be able to enter the phone number as the placeholder shows.

## Actual Result:
The user can not enter the phone number as the placeholder shows.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/2f41a1a2-5ff6-4063-bfc4-50c69258884f

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01f2cb151e80502fdb</li>
        <li>Upwork Job ID: 1765154798992896000</li>
        <li>Last Price Increase: 2024-03-05</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mallenexpensify</details>

**Comment by izarutskaya** [comment]
@mallenexpensify I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by dragnoir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Phone number in the format (440)458-9784 is marked as invalid

### What is the root cause of that problem?

Issue is within the function `isValidUSPhone`
https://github.com/Expensify/App/blob/578769197ddd4039f46b949d83a7e85ad2476abc/src/libs/ValidationUtils.ts#L281C10-L293

When the input is in phormat like: *(440)458-9784*
The condition below:
https://github.com/Expensify/App/blob/578769197ddd4039f46b949d83a7e85ad2476abc/src/libs/ValidationUtils.ts#L287-L289
will turn:
`regionCode:  US` and `!Str.isValidPhone(phone):  true`
Then the execution will exit without approving the phone number

### What changes do you think we should make in order to solve the problem?

We need to fix this condition
https://github.com/Expensify/App/blob/578769197ddd4039f46b949d83a7e85ad2476abc/src/libs/ValidationUtils.ts#L285-L289

It's built to fix an issue as mentioned in the comment:

> // When we pass regionCode as an option to parsePhoneNumber it wrongly assumes inputs like '=15123456789' as valid

we need to check if it's a valid phone number in another way.  We can create a js function `isValidPhoneNumber` if true then continue the execution, if not then exit.

### What alternative solutions did you explore? (Optional)

**Comment by mallenexpensify** [comment]
Was able to reproduce and seems like we'd want to fix. I, personally, would find it frustrating. 
Let me find which wave to drop it into. 
@Pujan92 , can you also please try to repro? Bank account deets [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1701353079991679?thread_ts=1701352541.652859&cid=C01GTK53T8Q)

**Comment by waleedj1** [comment]
This issue is due to Validation For Phone Number Input Field Is Failing, Due To Some Issue In The Regex, We can solve this by improving our regex

**Comment by waleedj1** [comment]
This issue is due to Validation For Phone Number Input Field Is Failing, Due To Some Issue In The Regex, We can solve this by improving our regex

**Contributor details**
Expensify Account: waleedjaved308@gmail.com
upwork Account: https://www.upwork.com/freelancers/~0124f16df0b524ada4

**Comment by mallenexpensify** [comment]
Thanks for testing @Pujan92 .  I've added this to the wave-collect project as Polish. 

> Do we need to allow users to enter in placeholder format((xxx) xxx-xxxx) or should it be like the way we allow for the login

Are there other examples of where a phone number is entered and accepted in multiple formats?  Ideally want input to be consistent with otheres (and... to also allow for multiple formats, if possible).

**Comment by Pujan92** [comment]
> > Do we need to allow users to enter in placeholder format((xxx) xxx-xxxx) or should it be like the way we allow for the login
> 
> Are there other examples of where a phone number is entered and accepted in multiple formats? Ideally want input to be consistent with otheres (and... to also allow for multiple formats, if possible).

Earlier it used to be working with this format(`(xxx) xxx-xxxx`) and many more. I think this problem arises after this [PR](https://github.com/Expensify/App/pull/35293). To allow multiple format case we need a solution which solves this issue as well as https://github.com/Expensify/App/issues/34766.

**Comment by dragnoir** [comment]
@Pujan92 my proposal allows multiple formats

**Comment by Pujan92** [comment]
Thanks @dragnoir , I will review the proposals tomorrow.

**Comment by Pujan92** [comment]
All proposals RCA are correct, I like the @tienifr's [proposal](https://github.com/Expensify/App/issues/37723#issuecomment-1978495658) as it is more detailed and agree with the fact that `isValidPhone` is used at multiple places and some places we are not supporting multiple formats(for eg. login). So renaming it to `isValidE164Phone` and creating other function for supporting multiple formats makes more sense to me.

🎀👀🎀 C+ reviewed

**Comment by dragnoir** [comment]
@Pujan92 I suggested these updates first. Please check again. I also was the 1st to figure out the RCA

**Comment by dragnoir** [comment]
@tienifr you changed everything after the discussion.

**Comment by Pujan92** [comment]
@dragnoir Regarding your proposed solution you have mentioned replacing `isValidPhone` with other function which accepts all formats. But we need this function also to support only E164 format. Also the selected updated proposal provided the regex which seems fine to me. 
Let's wait for @grgia's thoughts.

**Comment by dragnoir** [comment]
@Pujan92 there's a condition that checks if the phone number starts with = like (=5436789876) and turn false

it also turn false for numbers like (xxx) xxx-xxxx

Here I al the 1st who figured out this RCA.

Then I suggested fixing it so it catch format like (=xxxxxxx) as false and format like (xxx)xxx-xxxx as true

I also suggested to make the new function figure out all possible formats

**Comment by dragnoir** [comment]
@mallenexpensify and @grgia I am optimistic about your judicious decision-making. Please take a moment to review how significantly the proposal was transformed by tienifr, especially after mallenexpensifyr declined to simply adjust the placeholder. Additionally, it's important to note tienifr's mischaracterization of the Root Cause Analysis (RCA). He was not aware that the function isValidUSPhone is effectively operational when utilizing parsePhoneNumber, and the actual issue stemmed merely from a condition handling a specific case of a number format like xxxxxx. Furthermore, following the change, tienifr acknowledged the issue with isValidPhone, which was initially outlined in my proposal. This adjustment not only rectifies the oversight but also reinforces the original accuracy and foresight of my proposal.

**Comment by dragnoir** [comment]
Thank you @grgia for reviewing the proposals. I will take the $200.

**Comment by dragnoir** [comment]
@tienifr, I appreciate your effort in reviewing the pricing strategy; however, it seems there has been a misunderstanding. The entirety of the proceeds are allocated to me, which contradicts your Root Cause Analysis (RCA) and the subsequent proposal you made for overhauling the function or modifying the input placeholder. It's important to note that the focus of our discussion has shifted significantly towards the isValidPhone function, diverging from the original scope of your proposal.

I have agreed to the $200 arrangement as a gesture of respect for @grgia's decision.

**Comment by dragnoir** [comment]
@tienifr this is your initial proposal (updates later with my ideas)

![image](https://github.com/Expensify/App/assets/12425932/8edbb8ba-9d82-4c09-90e9-c4028ca07d28)

You said about the RCA:

> We are using isValidUSPhone to validate the phone number, and it doesn't consider the (xxx)xxx-xxx is the valid phone

and this is totally wrong. BECAUSE the function `isValidUSPhone` considers all formats of phone numbers because we are using `parsePhoneNumber` to check the phone format (we use the library awesome-phonenumber)

**So what you said is wrong**, isValidUSPhone consider the (xxx)xxx-xxx format.

In my proposal, I mentioned the correct RCA, we have a condition to turn false if the number is in this format (=xxxxxxxxxx) and suddenly this condition wrongly turn false for the format (xxx)xxx-xxx 
and **this is the right RCA**


Your solution was to fix isValidUSPhone with a new Regex and this is WRONG.

My solution was to fix the condition to only filter for mat like (=xxxxxxx) and not (xxx)xxx-xxxx and `this the best way to do it`

**Comment by dragnoir** [comment]
@tienifr my proposal and yours are totally different. You never mentioned isValidPhone. so recap:

- You said `isValidUSPhone` si wrong and we should fix it
==> WRONG

- I said `isValidUSPhone` is good and a condition inside it should be fixed
==> CORRECT

if your initial proposal was correct, why did you changed it totally with agreement and ideas from mine? Because you were pointing to the wrong RCA

**Comment by dragnoir** [comment]
@grgia BTW this is the second time @tienifr steal my idea, you can check here https://github.com/Expensify/App/issues/33709

**Comment by grgia** [comment]
@tienifr @dragnoir I'm going to discuss this internally and let @mallenexpensify give an update. 

For now, let's pause for a minute and wait for an internal update from the pending conversation.

We've got more than enough issues to go around and if you're having trouble finding work in the meantime, I'm happy to help you find something to work on.

**Comment by mallenexpensify** [comment]
👋 , here to help!  I don't have an immediate answer.  
The first piece I'm wondering about it - @tienifr when you were making edits to your proposal, for any code that would have affected the solution, did you also post a comment in the GH?  Asking because of the [below in CONTRIBUTING.md ](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#:~:text=If%20you%20want%20to%20make,Proposal%0A%5BUpdated%5D(link%20to%20proposal))
>  If you want to make an entirely new proposal or update an existing proposal, please go back and edit your original proposal, then post a new comment to the issue in this format to alert everyone that it has been updated:
```
## Proposal
[Updated](link to proposal)
```

**Comment by mallenexpensify** [comment]
Thanks @tienifr , I'm seeing that is from March 6, 8:42pm.
<img width="528" alt="image" src="https://github.com/Expensify/App/assets/22508304/f40850df-8a1a-4ec9-9233-0b528df69651">

> Ideally want input to be consistent with otheres (and... to also allow for multiple formats, if possible).

@dragnoir , are you specifically saying that you addressed this before March 6 (my time) as part of you proposal? 

Also.. let's all focus on getting the PR merged as we continue discussions about the process and potential compensation.

**Comment by dragnoir** [comment]
> > Ideally want input to be consistent with otheres (and... to also allow for multiple formats, if possible).
> 
> @dragnoir , are you specifically saying that you addressed this before March 6 (my time) as part of you proposal?
> 
> Also.. let's all focus on getting the PR merged as we continue discussions about the process and potential compensation.


@mallenexpensify Yes I addressed this before March 6. This is the original proposal from tienifr https://github.com/Expensify/App/issues/37723#issuecomment-1983272314 and he never mentioned that only after my proposal.
and this is what I mentioned before him:

> We can replace it with another function we create to check all possible format and return a true for valid US phone numbers.

@Pujan92 as mentioned above and here https://github.com/Expensify/App/issues/37723#issuecomment-1983288015 please be fair.

Ready to handle the RP.

**Comment by dragnoir** [comment]
Whatever the decision, @Pujan92 PR ready https://github.com/Expensify/App/pull/38028
(Working on screenshots) and I provided a list of tests I have done with random phone numbers.

**Comment by Pujan92** [comment]
@mallenexpensify After re-reviewing the proposals I still think @tienifr's proposal is more accurate here. 

Reason - In [expensify-common](https://github.com/Expensify/expensify-common/blob/77d0b150ba6bfbe7a64b3c3e30b65592b2e58c4a/lib/str.js#L934-L936) we have a util function called `isValidPhone` but it only considers the pattern for E164 format. So it is better to update that common lib with the 1) Rename it to the `isValidE164Phone` and 2) Create a new `isValidPhone` function to be more generic.

However, a detailed explanation has been added after @dragnoir's proposal and their proposal will solve the issue too. 

I would suggest assigning @tienifr here and best to split the bounty as @grgia suggested earlier($300 for @tienifr & $200 for @dragnoir). For this specific case, I expect both contributors to mutually agree and proceed to fix the issue. With that, we have many other external issues for which we are looking for proposals.

**Comment by dragnoir** [comment]
@Pujan92 I think you are talking about my proposal

> We can fix the function isValidPhone from expensify-common to consider the format (xxx)xxx-xxxx as a valid phone number

Where did you find that tienifr mentioned about fixing the common lib or creating new function? (below in screenshot of his proposal) All what he mentioned is fixing the main function `isValidUSPhone`

![image](https://github.com/Expensify/App/assets/12425932/4020f572-91f4-4199-8f00-ecf3d9ed5695)

**Comment by Pujan92** [comment]
> We can fix the function isValidPhone from expensify-common to consider the format (xxx)xxx-xxxx as a valid phone number

We can't directly add this format check to expensify-common `isValidPhone` function as we do need to support E164 format validation separately which is used in some places.

> Where did you find that tienifr mentioned about fixing the common lib or creating new function? (below in screenshot of his proposal) All what he mentioned is fixing the main function isValidUSPhone

@dragnoir I have mentioned it is added in their updated proposal after your proposal, That's why we are deciding on splitting the bounty. Let's wait now for @grgia and @mallenexpensify for the final decision and assignment.

**Comment by dragnoir** [comment]
Thanks god you finally admitted that everything about fixing the common lib or creating new function allow for multiple formats is added **AFTER** my proposal. 

@Pujan92 Where it's mentioned that updating a wrong proposal based on a second correct proposal leads to splitting the bounty. This is a violation of the contributions rules!!!!

**Comment by mallenexpensify** [comment]
Thanks for the patience here.  Coming from [here in CONTRIBUTING.md](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#:~:text=After%20you%20reproduce%20the%20issue). 
> After you reproduce the issue, complete the [proposal template here](https://github.com/Expensify/App/blob/main/contributingGuides/PROPOSAL_TEMPLATE.md) and post it as a comment in the corresponding GitHub issue (linked in the Upwork job).
Note: Before submitting a proposal on an issue, be sure to read any other existing proposals. ALL NEW PROPOSALS MUST BE DIFFERENT FROM EXISTING PROPOSALS. The difference should be important, meaningful or considerable.

Since the issue persists that it's possible that proposals are edited and updated from the time their first posted, we've ben working on `Proposal Police` to see if it can help with monitoring updates.  [From this post](https://github.com/Expensify/App/issues/35108#issuecomment-1983799697)

> - tienifr posted initial proposal - all good, no action performed
> - dragnoir posted initial proposal - all good, no action performed
> - dragnoir 1st update (added alternative solution) - minor proposal change detected, no action performed
> - dragnoir 2nd update - significant proposal change detected - proposal police edited the proposal adding the following line at the top: Edited by proposal-police: This proposal was edited at 2024-03-07 16:31:55 UTC.
> - tienifr 1st update - significant proposal change detected - proposal police edited the proposal adding the following line at the top: Edited by proposal-police: This proposal was edited at 2024-03-07 16:41:05 UTC.

So it looks like both proposals were significantly-enough updated to get to the point where @Pujan92 was able to review and recommend one.   

For me, being the BugZero team member who manages the issues, and as a non-engineer, it's hard to _judge_ the level of significance of the updates and changes.  It appears where we're here now
- @tienifr submits first proposal
- @dragnoir submits a proposal a few hours later
- Neither were _perfect_ and were updated
- @Pujan92 recommends @tienifr 
- @dragnoir believes that they provided useful data/code to help reach the solution and therefore should either be hired or compensated. 

Is the above correct?  If so, @tienifr can you provide reasoning why you think @dragnoir isn't due compensation or propose an amount you feel is fair for the split for their contributions?  Thx

**Comment by dragnoir** [comment]
@mallenexpensify After all the discussion and the analysis, it's now obvious that tienifr changed his proposal totally after me.
So, it's clear that I am the first one who created a clear and correct proposal. Also Pujan92 mentioned that his proposal is updated after mine.

> - dragnoir 2nd update - significant proposal change detected - proposal police edited the proposal adding the following line at the top: Edited by proposal-police: This proposal was edited at 2024-03-07 16:31:55 UTC.
> - tienifr 1st update - significant proposal change detected - proposal police edited the proposal adding the following line at the top: Edited by proposal-police: This proposal was edited at 2024-03-07 16:41:05 UTC.

> @dragnoir I have mentioned it is added in their updated proposal after your proposal, 

For that, **I will not accept the split**. I am sure that I am eligible for the full compensation. We have rules to follow and guides to respect. We need to stop all those attempts to frustrate the decision by posting the wrong proposal quickly to get first place and come back later to fix it. My proposal is the first one with the correct RCA and clear fixes.

Thank you and I will respect your decision.

**Comment by mallenexpensify** [comment]
@Pujan92 can you please review @tienifr 's updates to their proposal [here](https://github.com/Expensify/App/issues/37723#issuecomment-1978495658) to see if any of the changes made in the proposal included code that @dragnoir had already included in any of their versions of their proposal before @tienifr 's updates were made?  If so, can you include those in a post?  Thx

**Comment by Pujan92** [comment]
@mallenexpensify Though @dragnoir's alternative solution is on the right track but fails to solve all cases and can cause regressions, I don't see much similarity in @tienifr's updated proposal with it.

**Comment by Pujan92** [comment]
> We need to stop all those attempts to frustrate the decision by posting the wrong proposal quickly to get first place and come back later to fix it. 

I agree with this @dragnoir, I believe the contributor should post a new proposal for the significant changes as it is counted as a new proposal only.

**Comment by dragnoir** [comment]
> I don't see much similarity in @tienifr's updated proposal with it.

@Pujan92 below a screenshot comparing my proposal (posted first) with the updated proposal from tienifr

as noted on the screenshot:
1- He changed his proposal to talk about a new RCA, he was talking about the wrong function `isValidUSPhone` and then he changed it to talk about the condition inside it `isValidPhone`.
2- He was not aware that the condition within the function is failing and that's the RCA, then he edited his proposal to mention that part of the code
3- From the 1st proposal, before he update it, he was thinking that `isValidUSPhone` doesn't consider the format (xxx)xxx-xxxx then he changed it to mention that the condition fails for this type of phone number.

![image](https://github.com/Expensify/App/assets/12425932/2be0f2e7-a6e7-42af-b36f-45e244865d10)

Now about the solution, please again check the screenshot below:

![image](https://github.com/Expensify/App/assets/12425932/6e7a1b0d-c8be-4613-a147-7aeefd581302)

I am the first who mentioned to **CREATE ANOTHER FUNCTION TO HANDLE ALL POSSIBLE PHONE NUMBERS**

> Though @dragnoir's alternative solution is on the right track but fails to solve all cases

Before my alternative solution, the solution I provided solve all the cases. Why you are talking about the alternative solution when the proposal solution is fixing all the issue!!!!!


> I agree with this @dragnoir, I believe the contributor should post a new proposal for the significant changes as it is counted as a new proposal only.

Sorry but I don't see any significant changes on tienifr updates. What I see is all a copy of mine.

@Pujan92 Can you please review again, hope those screenshots gives you a clear picture and more easy way to understand the points I mentioned above.

**Comment by dragnoir** [comment]
@mallenexpensify I created a post above with details and screenshots to make it clear how similar my proposal is to the update.

**Comment by mallenexpensify** [comment]
Apologies for the delay, we're [discussing internally,](https://expensify.slack.com/archives/C03TQ48KC/p1710892644836699) I'm trying to find an engineer who's good at comparing proposal edits to give us a more definitive answer.

**Comment by mallenexpensify** [comment]
OK.. I'm back!  I put the proposals into a google doc to compare them. It's [here](https://docs.google.com/document/d/1f7H3U1TG1TrvHb5mCeQBJDxEfJoIwxs4XTLD70VVAao/edit?usp=sharing)

@tienifr was the first to note the bug was to state the root cause of the problem, **[in their first unedited proposal](https://github.com/Expensify/App/issues/37723#issuecomment-1978495658)**,  was: "We are using [isValidUSPhone](https://github.com/Expensify/App/blob/9c49ab65db3f40e719c75516d5655f11b68c0f87/src/libs/ValidationUtils.ts#L281) to validate" .  In the same sentence they also state "it doesn't consider the (xxx)xxx-xxx is the valid phone" 

@tienifr did not mention the below til after @dragnoir 's proposal.  
https://github.com/Expensify/App/blob/578769197ddd4039f46b949d83a7e85ad2476abc/src/libs/ValidationUtils.ts#L287-L289

@Pujan92 , how important to the proposal do you think the above is?

**Comment by mallenexpensify** [comment]
@tienifr , this is not a high priority or time sensitive bug.  We've been needing to understand how best to address quickly submitted proposals that are then updated after.  It's become a _real problem_ that had been addressed by many contributors.  My goal is to help create a process, partially working off this issue as an example.  Hang tight and we'll have an update soon.

**Comment by Pujan92** [comment]
> https://github.com/Expensify/App/blob/578769197ddd4039f46b949d83a7e85ad2476abc/src/libs/ValidationUtils.ts#L287-L289
> 
> @Pujan92 , how important to the proposal do you think the above is?

@mallenexpensify It is important because the issue exists after the addition of `Str.isValidPhone` part.

**Comment by mallenexpensify** [comment]
I was just [reviewing CONTRIBUTING.md](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#:~:text=Note%3A%20Before%20submitting%20a%20proposal%20on%20an%20issue%2C%20be%20sure%20to%20read%20any%20other%20existing%20proposals.%20ALL%20NEW%20PROPOSALS%20MUST%20BE%20DIFFERENT%20FROM%20EXISTING%20PROPOSALS.%20The%20difference%20should%20be%20important%2C%20meaningful%20or%20considerable.)
> Note: Before submitting a proposal on an issue, be sure to read any other existing proposals. ALL NEW PROPOSALS MUST BE DIFFERENT FROM EXISTING PROPOSALS. The difference should be important, meaningful or considerable.

Based on @Pujan92 's comment above
> It is important because the issue exists after the addition of Str.isValidPhone part.

It's seems pretty clear to me that @tienifr had the RCA correct first and @dragnoir followed with an important improvement that wasn't part of @tienifr 's initial proposal.  If we can agree on that, I don't understand why @tienifr and @dragnoir are both stating that a split isn't fair.  I also agree with what @grgia [proposed two weeks ago](https://github.com/Expensify/App/issues/37723#issuecomment-1983220475) and that @Pujan92 also agreed with
> How about we split the bounty for this issue? $200 / $300 for @tienifr to open the PR/do screenshots/testing as I trust @Pujan92's C+ decision.

**Comment by mallenexpensify** [comment]
Thanks all!  @tienifr please create the PR.

**Comment by Pujan92** [comment]
@tienifr In the regex we forgot to add starts with(^) and ends with($) character which seems to cause a regression here. Thanks @dragnoir and wondering how we missed this in https://github.com/Expensify/App/pull/38993#issuecomment-2021327933 and sorry it was missed in my testing too. @tienifr Can you plz check and raise a follow-up PR in expensify-common? 
cc: @grgia

**Comment by Pujan92** [comment]
> I think it's better to check if the phone starts with `=` and let the `parsePhoneNumber` do the check.

I don't think we should do like that as `parsePhoneNumber` will consider the string valid even though it contains `=`. To avoid this only we have made the other check with https://github.com/Expensify/App/pull/35293. That's why we need to validate numbers format instead of relying on the `parsePhoneNumber`.

**Comment by Pujan92** [comment]
I believe adding a start and end char in the regex is a cleaner way here. I will raise a PR for the fix in case @tienifr isn't available shortly.

**Comment by mallenexpensify** [comment]
Checking on something, will issue payment shortly
@tienifr , @Pujan92 @dragnoir , can you please apply to the job and reply here once you have. Thx
https://www.upwork.com/jobs/~01b909addf4ed7ece2

**Comment by mallenexpensify** [comment]
@Pujan92 , @tienifr , @dragnoir can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~01b909addf4ed7ece2

**Comment by mallenexpensify** [comment]
Contributor: @dragnoir paid $200 via Upwork
Contributor: @tienifr paid $300 via Upwork
Contributor+: @Pujan92 paid $250 via Upwork (reflects deduction from regression).  

**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

@Pujan92 , can you please fill out the BZ checklist? One wasn't auto posted, so pasting below. 

- [ ] [@] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@] Determine if we should create a regression test for this bug.
- [ ] [@] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@BZ] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:
- [ ]

**Comment by Pujan92** [comment]
**BugZero Checklist**: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [X] [@Pujan92] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/35293
- [X] [@Pujan92] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/35293#discussion_r1568168949
 - [X] [@Pujan92] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: NA
 - [X] [@Pujan92] Determine if we should create a regression test for this bug. Yes
 - [X] [@Pujan92] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.
 
### Regression Test Steps

1. Go to Workspace > Bank account.
2. Initiate the add BA flow and reach to the Company information screen.
3. Try to enter phone number with the format (xxx)xxx-xxxx 
4. Verify the number gets validated and won't show any error

Do you agree 👍 or 👎

**Comment by mallenexpensify** [comment]
Thanks @Pujan92 
TR GH is here
- https://github.com/Expensify/Expensify/issues/392197
